src/OpenDsc.Resource.Windows.User/Utils.cs

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) Thomas Nieto - All Rights Reserved
+// You may use, distribute and modify this code under the
+// terms of the MIT license.
+
+using System.DirectoryServices.AccountManagement;
+using OpenDsc.Resource;
+
+namespace OpenDsc.Resource.Windows.User;
+
+internal static class Utils
+{
+    public static Schema GetUser(string userName)
+    {
+        try
+        {
+            using var context = new PrincipalContext(ContextType.Machine);
+            using var user = UserPrincipal.FindByIdentity(context, userName);
+
+            return user == null
+                ? new Schema { userName = userName, exist = false }
+                : new Schema
+                {
+                    userName = userName,
+                    exist = true,
+                    fullName = user.DisplayName,
+                    description = user.Description,
+                    disabled = !user.Enabled,
+                    passwordNeverExpires = user.PasswordNeverExpires,
+                    passwordChangeNotAllowed = user.UserCannotChangePassword,
+                    passwordChangeRequired = IsPasswordChangeRequired(user)
+                };
+        }
+        catch (Exception ex)
+        {
+            throw new InvalidOperationException($"Failed to retrieve user '{userName}': {ex.Message}", ex);
```

Comment:
You shouldn't need to catch this exception as the library will catch it for you. If the called methods return specific exceptions then put those into the exit codes.